cmd/agent/subcommands/run/dependent_services.go

```diff
@@ -36,3 +41,19 @@ func startDependentServices() {
 		}
 	}
 }
+
+func stopDependentServices() {
+	for _, svc := range subservices {
+		if svc.IsEnabled() && svc.ShouldStop() {
+			log.Debugf("Attempting to stop service: %s", svc.name)
+			err := svc.Stop()
+			if err != nil {
+				log.Warnf("Failed to stop services %s: %s", svc.name, err.Error())
+			} else {
+				log.Debugf("Stopped service %s", svc.name)
+			}
+		} else {
+			log.Infof("Service %s is disabled, not stopping", svc.name)
```

Comment:
This log message will be deceptive in the case that the service is enabled but `ShouldStop` returns false. Maybe instead:

```suggestion
		} else if svc.IsEnabled() {
			log.Infof("Service %s is enabled and doesn't need to be stopped", svc.name)
		} else {
			log.Infof("Service %s is disabled, not stopping", svc.name)
```